db/crud.go

```diff
@@ -355,11 +355,19 @@ func (db *Database) GetDelta(docID, fromRevID, toRevID string) (delta *RevisionD
 			return nil, nil, err
 		}
 
-		isAuthorized, redactedBody := db.authorizeUserForChannels(docID, toRevID, toRevision.Channels, false, toRevision.History)
+		deleted, _ := toRevision.Body[BodyDeleted].(bool)
+		isAuthorized, redactedBody := db.authorizeUserForChannels(docID, toRevID, toRevision.Channels, deleted, toRevision.History)
 		if !isAuthorized {
 			return nil, redactedBody, nil
 		}
 
+		// If the revision we're generating a delta to is a tombstone, mark it as such and don't bother generating a delta
+		if deleted {
+			revCacheDelta := newRevCacheDelta([]byte(`{}`), fromRevID, toRevision, deleted)
+			db.revisionCache.UpdateDelta(docID, fromRevID, revCacheDelta)
+			return revCacheDelta, nil, nil
+		}
```

Comment:
Do the unit tests exercise the case where an initial request generates the empty delta with deleted=true, and then a subsequent request retrieves that delta from the cache?  The logic looks sound, but would be good to have it validated.

Reply:
I should be able to get a test to force that to happen yeah. In the real world CBL shouldn't be asking for a delta from a tombstoned revision, but I'll update the test tomorrow anyway to get that covered.

Reply:
I just meant a delta *to* a tombstoned revision.  Basically the same case that TestBlipDeltaSyncPullTombstoned covers, but where the (empty) delta with deleted=true is already resident in the cache from a previous request.